AdminCore.WebApp/Controllers/EventController.cs

```diff
@@ -0,0 +1,276 @@
+      if (events != null)
+      {
+        return Ok(_mapper.Map<IList<EventViewModel>>(events));
+      }
+      return StatusCode((int)HttpStatusCode.NoContent, "No Event exists");
+    }
+
+    [HttpGet("findEventMessages/{eventId}")]
+    public IActionResult GetEventMessages(int eventId)
+    {
+      var eventMessages = _eventMessageService.GetAllEventMessagesForEvent(eventId);
+      if (eventMessages != null)
+      {
+        return Ok(_mapper.Map<IList<EventMessageViewModel>>(eventMessages));
+      }
+
+      return StatusCode((int)HttpStatusCode.NoContent, "No Messages exists");
+    }
+
+    [HttpGet("findEmployeeHolidayStats/{employeeId}")]
+    public IActionResult GetEmployeeHolidayStats(int employeeId)
+    {
+      var holidayStats = _eventService.GetHolidayStatsForUser(employeeId);
+      if (holidayStats != null)
+      {
+        return Ok(_mapper.Map<HolidayStatsViewModel>(holidayStats));
+      }
+
+      return StatusCode((int)HttpStatusCode.NoContent, "No Holiday exists");
+    }
+
+    [HttpPost]
+    public IActionResult CreateEvent(CreateEventViewModel createEventViewModel)
+    {
+      if (createEventViewModel.EventTypeId == 0)
+      {
+        return StatusCode((int)HttpStatusCode.InternalServerError, "You may not create new public holidays");
+      }
+      var eventDates = _mapper.Map<EventDateDto>(createEventViewModel);
+      try
+      {
+        if (IsHolidayEvent(createEventViewModel.EventTypeId))
+        {
+          _eventService.IsEventValid(eventDates, createEventViewModel.IsHalfDay, _employee.EmployeeId);
+        }
+        _eventService.CreateEvent(eventDates, (EventTypes)createEventViewModel.EventTypeId, _employee.EmployeeId);
+        return Ok($"Event has been created successfully");
+      }
+      catch (Exception ex)
+      {
+        Logger.LogError(ex.Message);
+        return StatusCode((int)HttpStatusCode.InternalServerError, "Error creating event: " + ex.Message);
+      }
+    }
+
+    [HttpPut]
+    public IActionResult UpdateEvent(UpdateEventViewModel updateEventViewModel)
+    {
+      var eventDatesToUpdate = _mapper.Map<EventDateDto>(updateEventViewModel);
+      try
+      {
+        _eventService.IsEventValid(eventDatesToUpdate, updateEventViewModel.IsHalfDay, _employee.EmployeeId);
+        _eventService.UpdateEvent(eventDatesToUpdate, updateEventViewModel.Message, _employee.EmployeeId);
+        return Ok("Holiday has been successfully updated");
+      }
+      catch (Exception ex)
+      {
+        Logger.LogError(ex.Message);
+        return StatusCode((int)HttpStatusCode.InternalServerError, "Error updating holiday: " + ex.Message);
+      }
+    }
+
+    [Authorize("Admin")]
+    [HttpPut("approveEvent")]
+    public IActionResult ApproveEvent(ApproveEventViewModel approveEventViewModel)
+    {
+      try
+      {
+        var eventToApprove = _eventService.GetEvent(approveEventViewModel.EventId);
+        if (eventToApprove.EmployeeId != _employee.EmployeeId)
```

Comment:
factor this admin check out into it's own method and make it throw an exception if an admin tries to approve their own holiday. This will be caught and returned by the catch below.